### DISCUSSION TITLE
Firebase Message not delivering or delivering late to Android due to battery optimization

### DISCUSSION THREAD

**Discussion by cowera** [kreait/firebase-php#547] [General]
My code:

```
$messaging = app('firebase.messaging');
        $message = CloudMessage::fromArray([
            'token' => $token,
            'priority' => "high",
            'notification' => $notification,
            'data' => $data
        ]);
$messaging->send($message)
```

My users complain that they don't receive the messages or they get the messages too late. For my use case it's important that message are being delivered asap. Any idea what to do? Thanks!

**Comment by jeromegamez** [comment]
Unfortunately, there's nothing we can do from a sender's perspective. Sending the message to Firebase is fast: as long as you don't get an error back, we have to assume that everything is going well so far. Message *delivery*, on the other hand, is out of our control. I've received many reports on slow or failed message delivery, and every time, the situation got better after a certain amount of time. You should keep an eye on https://status.firebase.google.com/ , and if things don't get better, open a support ticket with [Firebase support](https://firebase.google.com/support)
